### PR TITLE
feat(config): beginning of k8s provider

### PR DESCRIPTION
**Pull request by tyler-dunkel**
This PR adds several things to the K8s provider and sets up how how it works.

* basic enums and boilerplate
* config function for cli
* authentication mechanism
* Node service
* Namespace service

### CODE REVIEW

File: src/services/index.ts [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+  }
+
+  async configure(): Promise<{ [key: string]: any }> {
+    const result: { [key: string]: any } = {}
+    const { flags = {} } = this.config
+    const kc = new k8s.KubeConfig()
+    kc.loadFromDefault()
+    const contextList = kc.getContexts()
+    kc.setCurrentContext(
+      'arn:aws:eks:us-east-1:632941798677:cluster/autocloud-sandbox-eks'
```
Comment (kunovsky) [inline src/services/index.ts]: We probably don't want to include this

File: src/services/index.ts [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    // client.listNamespace
+    // client.listPodForAllNamespaces
+    // client.listPodTemplateForAllNamespaces
+    // client.listReplicationControllerForAllNamespaces
+    // client.listResourceQuotaForAllNamespaces
+    try {
+      for (const resource of resourceNames) {
+        const serviceClass = this.getService(resource)
+        if (serviceClass && serviceClass.getData) {
+          // const startDate = new Date()
```
Comment (kunovsky) [inline src/services/index.ts:170]: needed?

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+
+    /* Basic Options */
```
Comment (kunovsky) [inline tsconfig.json:5]: remove the commented-out lines for simplicity?

**Review by kunovsky — COMMENTED** [review]
LGTM, just a few comments

**Comment by hjaraujof** [comment]
Hey @tyler-dunkel, this looks good. I have two questions: 

- Do you plan to add some query filters for the `k8sNamespace` and `k8sNode` types? 
- Do you plan to use a base type to prevent repeating `id`, `context`, `apiVersion`, `kind` properties for each service(and type/subtype) added to the k8 provider?

File: tsconfig.json (hunk 3 above)
Comment (tyler-dunkel) [reply tsconfig.json:5]: I think ill keep these as its generated by the ts init config. We may want to update in the future.

File: src/services/index.ts (hunk 2 above)
Comment (tyler-dunkel) [reply src/services/index.ts:170]: yes will be using later down the road

File: src/services/index.ts (hunk 1 above)
Comment (tyler-dunkel) [reply src/services/index.ts]: yep will remove that

**Review by hjaraujof — CHANGES_REQUESTED** [review]
Hey @tyler-dunkel, this looks good. I have two questions: 

- Do you plan to add some query filters for the `k8sNamespace` and `k8sNode` types? 
- Do you plan to use a base type to prevent repeating `id`, `context`, `apiVersion`, `kind` properties for each service(and type/subtype) added to the k8 provider?

**Comment by tyler-dunkel** [comment]
@hjaraujof I will be added the search and generate directives. I dont think ill be using a base type until I see more of the services and their structure.

File: src/services/namespace/schema.graphql [hunk 4]
```diff
@@ -0,0 +1,18 @@
+type k8sNamespace {
+  id: String! @id
+  context: String!
+  apiVersion: String
+  kind: String
```
Comment (hjaraujof) [inline src/services/namespace/schema.graphql]: This could be part of a BaseK8Resource type

Comment (hjaraujof) [inline src/services/namespace/schema.graphql]: `id`, `context`, `apiVersion`, `kind` could be part of a BaseK8Resource type

File: src/services/node/schema.graphql [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  kind: String
+  hostName: String
+  externalIp: String
+  internalIp: String
+  metadata: k8sMetadata
+  spec: k8sNodeSpec
+  status: k8sNodeStatus
+}
+
+type k8sMetadata {
```
Comment (hjaraujof) [inline src/services/node/schema.graphql]: This type could also live in the base schema file for the provider

**Review by hjaraujof — APPROVED** [review]

**Pull request merged**